apps/parsley/src/hooks/useSections/utils.ts

```diff
@@ -41,68 +56,97 @@ interface SectionEntry {
  * @returns The updated accumulated section data after processing the current line.
  */
 const reduceFn = (
-  accum: SectionEntry[],
+  accum: SectionData,
   line: string,
   logIndex: number,
-): SectionEntry[] => {
+): SectionData => {
   const currentLine = processLine(line);
   // Skip if the current line does not indicate a section
   if (!currentLine) {
     return accum;
   }
-  const sections = [...accum];
+  const { commands, functions } = {
+    commands: [...accum.commands],
+    functions: [...accum.functions],
+  };
   if (currentLine.status === SectionStatus.Finished) {
-    if (sections.length === 0) {
+    if (functions.length === 0 || commands.length === 0) {
       throw new Error(
         "Log file is showing a finished section without a running section before it.",
       );
     }
     // Update the end line number exclusive of the last section in the accumulator
-    sections[sections.length - 1].range.end = logIndex + 1;
-    return sections;
+    functions[functions.length - 1].range.end = logIndex + 1;
+    commands[commands.length - 1].range.end = logIndex + 1;
+    return { commands, functions };
   }
 
-  const isNewSection =
-    sections.length === 0 ||
-    sections[sections.length - 1].functionName !== currentLine.functionName;
   /**
-   * @description - ONGOING_SECTION is used to indicate that the section is still running and has not finished yet in the log file.
+   * @description - ONGOING_ENTRY is used to indicate that the section or command is still running and has not finished yet in the log file.
    * The section parsing function will temporarily assign this value until the corresponding finished section or EOF is found.
    */
-  const ONGOING_SECTION = -1;
+  const ONGOING_ENTRY = -1;
```

Comment:
[nit] Imo no need to use top-level const syntax here
```suggestion
  const ongoingEntry = -1;
```

Reply:
I prefer the uppercase const because it makes the code easier to read (for me). This style is accepted by [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const#basic_const_usage)